bpf/bpf_lxc.c

```diff
@@ -651,11 +658,18 @@ static inline int handle_ipv4_from_lxc(struct __sk_buff *skb, __u32 *dstID)
 		key.ip4 = orig_dip & IPV4_MASK;
 		key.family = ENDPOINT_KEY_IPV4;
 
-		ret = encap_and_redirect(skb, &key, SECLABEL, monitor);
-
-		/* Fall through if remote prefix was not found
-		 * (DROP_NO_TUNNEL_ENDPOINT) */
-		if (ret != DROP_NO_TUNNEL_ENDPOINT)
+		/* Three cases exist here either (a) the encap and redirect could
+		 * not find the tunnel so pass to host for further processing, (b)
+		 * the packet needs further stack processing likely due to IPSec so
+		 * pass up to stack or (c) packet was redirected to tunnel device
+		 * so return.
+		 */
+		ret = encap_and_redirect(skb, &key, SECLABEL, monitor, false);
+		if (ret == DROP_NO_TUNNEL_ENDPOINT)
+			goto to_host;
```

Comment:
```
#ifdef ENCAP_IFINDEX
...
        } else {
...
                ret = encap_and_redirect(skb, &key, SECLABEL, monitor);
    
                /* Fall through if remote prefix was not found
                 * (DROP_NO_TUNNEL_ENDPOINT) */
                if (ret != DROP_NO_TUNNEL_ENDPOINT)
                        return ret;
        }
#endif 
        goto pass_to_stack;

to_host:
```
so seems to_host -> pass_to_stack should fix it. Good catch.

```
                ret = encap_and_redirect(skb, &key, SECLABEL, monitor, false);
                if (ret == DROP_NO_TUNNEL_ENDPOINT)
                        goto pass_to_stack;
                else if (!ret)
                        goto pass_to_stack;
                else
                        return ret;
```

Reply:
how about above ^^^^ this introduces !ret as the IPSEC case basically. Could give it a naem I guess, IPSEC_TUNNEL_ENDPOINT so that its a bit cleaner,
```
else if (ret == IPSEC_TUNNEL_ENDPOINT)
```